roqoqo-iqm/src/interface.rs

```diff
@@ -62,51 +64,51 @@ pub struct IqmInstruction {
 // HashMap that associates to each register name the indices in the register that are being affected
 // by measurements. These indices are saved in the order in which the measurement operations appear
 // in the circuit, since this is the order in which the backend returns the results.
-pub(crate) type RegisterMapping = HashMap<String, Vec<usize>>;
+pub(crate) type MeasuredQubitsMap = HashMap<String, (Vec<usize>, usize)>;
 
-/// Converts all operations in a [roqoqo::Circuit] into instructions for IQM Hardware or IQM Simulators
+/// Converts all operations in a [roqoqo::Circuit] into instructions for IQM Hardware.
 ///
 /// # Arguments
 ///
 /// * `circuit` - The [roqoqo::Circuit] that is converted
-/// * `device_number_qubits` - The number of qubits of the backend device. It is used to know how many qubits to measure with [roqoqo::operations::PragmaRepeatedMeasurement]
-/// * `output_registers` - A mutable reference to the classical registers that need to be initialized
```

Comment:
Should this mention of PragmaRepeatedMeasurement be changed to PragmaSetNumberMeasurements/MeasureQubit?

Reply:
Why do you think so? We only need the info about the number of qubits of the device when we want to measure all of them, which is only the case with PragmaRepeatedMeasurement